kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/HasMetadataOperation.java

```diff
@@ -140,6 +140,9 @@ protected T replace(T item, boolean status) {
     String fixedResourceVersion = getResourceVersion();
     Exception caught = null;
     int maxTries = 10;
+    if (item.getMetadata() == null) {
+      item.setMetadata(new ObjectMeta());
+    }
```

Comment:
Sorry if I'm missing context. Why is this change required?

Reply:
Just to avoid the later npe - but it could be handled via an arg check or left out if you want.